instrumented/nightly-tests/src/androidTest/kotlin/com/datadog/android/nightly/logs/GdprLogsE2ETests.kt

```diff
@@ -0,0 +1,201 @@
+    val forge = ForgeRule()
+
+    @Before
+    fun setup() {
+        stopSdk()
+    }
+
+    // region Tests
+
+    /**
+     * apiMethodSignature: Credentials#constructor(String, String, String, String?, String? = null)
+     */
+    @Test
+    fun logs_credentials_consent_pending() {
+        val testMethodName = "logs_config_consent_pending"
+        measureSdkInitialize {
+            initializeSdk(
+                InstrumentationRegistry.getInstrumentation().targetContext,
+                consent = TrackingConsent.PENDING
+            )
+        }
+        measureLoggerInitialize {
+            logger = initializeLogger()
+        }
+        sendRandomLog(
+            forge.anAlphabeticalString(),
+            testMethodName,
+            forge.aNullable { forge.aThrowable() }
+        )
+    }
+
+    /**
+     * apiMethodSignature: Credentials#constructor(String, String, String, String?, String? = null)
+     */
+    @Test
+    fun logs_credentials_consent_granted() {
+        val testMethodName = "logs_config_consent_granted"
+        measureSdkInitialize {
+            initializeSdk(
+                InstrumentationRegistry.getInstrumentation().targetContext,
+                consent = TrackingConsent.GRANTED
+            )
+        }
+        measureLoggerInitialize {
+            logger = initializeLogger()
+        }
+        sendRandomLog(
+            forge.anAlphabeticalString(),
+            testMethodName,
+            forge.aNullable { forge.aThrowable() }
+        )
+    }
+
+    /**
+     * apiMethodSignature: Credentials#constructor(String, String, String, String?, String? = null)
+     */
+    @Test
+    fun logs_credentials_consent_not_granted() {
+        val testMethodName = "logs_config_consent_not_granted"
+        measureSdkInitialize {
+            initializeSdk(
+                InstrumentationRegistry.getInstrumentation().targetContext,
+                consent = TrackingConsent.NOT_GRANTED
+            )
+        }
+        measureLoggerInitialize {
+            logger = initializeLogger()
+        }
+        sendRandomLog(
+            forge.anAlphabeticalString(),
+            testMethodName,
+            forge.aNullable { forge.aThrowable() }
+        )
+    }
+
+    /**
+     * apiMethodSignature: Datadog#fun setTrackingConsent(com.datadog.android.privacy.TrackingConsent)
+     */
+    @Test
+    fun logs_credentials_consent_pending_to_granted() {
+        val testMethodName = "logs_config_consent_pending_to_granted"
+        measureSdkInitialize {
+            initializeSdk(
+                InstrumentationRegistry.getInstrumentation().targetContext,
+                consent = TrackingConsent.PENDING
+            )
+        }
+        measureLoggerInitialize {
+            logger = initializeLogger()
+        }
+        sendRandomLog(
+            forge.anAlphabeticalString(),
+            testMethodName,
+            forge.aNullable { forge.aThrowable() }
+        )
+        measureSetTrackingConsent {
```

Comment:
why do we need to measure things above and send the logs? just measuring consent is not enough?

Reply:
because we agreed in the RFC that we will test the SDK initialization and Logger initialization whenever we will have the chance. The SDK can be initialized in different ways and this way we can measure all the scenarios. On the other hand there is only one monitor : `sdk_initialize` and `logs_logger_initialize` in the Dashboard that checks if the execution time is lower than 16 ms.